code_examples/model_registration_pipeline_example.py

```diff
@@ -0,0 +1,181 @@
+        registered_model_and_version = registry.register_model(
+            name=model_name,
+            uri=model_artifact_uri,
+            description=model_description,
+            model_format_name="custom-format", # Example: "tensorflow", "pytorch", "onnx", "scikit-learn"
+            model_format_version="1.0",        # Version of the model format itself
+            version=model_version_name,        # The specific model version string (e.g., "v1.0.0")
+            owner="Kubeflow Pipelines",        # Owner of the model (e.g., team name)
+            author=model_author,               # Author of this specific model version
+            metadata={                         # Custom metadata about the model/training
+                "training_epochs": 100,
+                "accuracy": 0.95,
+                "pipeline_run_id": pipeline_run_id,
+                "pipeline_name": pipeline_name,
+                "pipeline_namespace": pipeline_namespace,
+                "kfp_component_name": "register-model-to-kubeflow-registry"
+            },
+            # Best practices for cross-referencing model source (from KEP):
+            model_source_id=pipeline_run_id,           # ID of the KFP pipeline run
+            model_source_name=pipeline_name,           # Name of the KFP pipeline
+            model_source_class="pipelinerun",          # KFP-specific identifier for run
+            model_source_kind="kfp",                   # KFP-specific identifier for source system
+            model_source_group=pipeline_namespace,     # Namespace where the pipeline ran
+        )
+        print(f"Registered Model ID: {registered_model_and_version.id}, Name: {registered_model_and_version.name}")
+        # The object returned might be a RegisteredModel or ModelVersion depending on client version.
+        # Assuming it's RegisteredModel, we can infer the version was created.
+
+        print(f"Successfully registered model and version in Model Registry.")
+        
+        # --- NEW: Set metadata on the KFP output model artifact ---
+        # Write some dummy content to the output model path
+        with open(output_model.path, 'w') as f:
+            f.write(f"This is a fake model artifact for {model_name} version {model_version_name}.")
+        
+        # Set metadata on the KFP output model artifact
+        output_model.metadata["registered_model"] = {
+            "modelName": registered_model_and_version.name,
+            "versionName": model_version_name, # Use the version name passed to component
+            "modelID": registered_model_and_version.id,
+            # In a real scenario, you'd get the actual modelVersionId from the API response
+            # For now, we'll use a placeholder or query it if needed.
+            "versionID": "placeholder-version-id", # Model Registry client doesn't return version ID directly from register_model
```

Comment:
You can just call `model_version_id = registry.get_model_version(model_name, model_version_name).id` to get it.